crates/iota-sdk-ffi/src/types/type_tag.rs

```diff
@@ -1,5 +1,198 @@
 // Copyright (c) 2025 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use std::sync::Arc;
+
+/// Type of a move value
+///
+/// # BCS
+///
+/// The BCS serialized form for this type is defined by the following ABNF:
+///
+/// ```text
+/// type-tag = type-tag-u8 \
+///            type-tag-u16 \
+///            type-tag-u32 \
+///            type-tag-u64 \
+///            type-tag-u128 \
+///            type-tag-u256 \
+///            type-tag-bool \
+///            type-tag-address \
+///            type-tag-signer \
+///            type-tag-vector \
+///            type-tag-struct
+///
+/// type-tag-u8 = %x01
+/// type-tag-u16 = %x08
+/// type-tag-u32 = %x09
+/// type-tag-u64 = %x02
+/// type-tag-u128 = %x03
+/// type-tag-u256 = %x0a
+/// type-tag-bool = %x00
+/// type-tag-address = %x04
+/// type-tag-signer = %x05
+/// type-tag-vector = %x06 type-tag
+/// type-tag-struct = %x07 struct-tag
+/// ```
 #[derive(Clone, Debug, derive_more::From, uniffi::Object)]
 pub struct TypeTag(pub iota_types::TypeTag);
+
+#[uniffi::export]
+impl TypeTag {
+    #[inline]
+    pub fn is_u8(&self) -> bool {
+        self.0.is_u8()
+    }
+
+    #[inline]
+    pub fn is_u16(&self) -> bool {
+        self.0.is_u16()
+    }
+
+    #[inline]
+    pub fn is_u32(&self) -> bool {
+        self.0.is_u32()
+    }
+
+    #[inline]
+    pub fn is_u64(&self) -> bool {
+        self.0.is_u64()
+    }
+
+    #[inline]
+    pub fn is_u128(&self) -> bool {
+        self.0.is_u128()
+    }
+
+    #[inline]
+    pub fn is_u256(&self) -> bool {
+        self.0.is_u256()
+    }
+
+    #[inline]
+    pub fn is_bool(&self) -> bool {
+        self.0.is_bool()
+    }
+
+    #[inline]
+    pub fn is_address(&self) -> bool {
+        self.0.is_address()
+    }
+
+    #[inline]
+    pub fn is_signer(&self) -> bool {
+        self.0.is_signer()
+    }
+
+    #[inline]
+    pub fn is_vector(&self) -> bool {
+        self.0.is_vector()
+    }
+
+    #[inline]
+    pub fn as_vector_type_tag_opt(&self) -> Option<Arc<TypeTag>> {
+        self.0
+            .as_vector_type_tag_opt()
+            .cloned()
+            .map(Into::into)
+            .map(Arc::new)
+    }
+
+    #[inline]
+    pub fn as_vector_type_tag(&self) -> TypeTag {
+        self.0.as_vector_type_tag().clone().into()
+    }
+
+    #[inline]
+    pub fn into_vector_type_tag_opt(&self) -> Option<Arc<TypeTag>> {
```

Comment:
No point in having this one since it's the same as the `as_` version

Reply:
Oh right, removed in https://github.com/iotaledger/iota-rust-sdk/pull/83/commits/a303b3e6715f72f41fc89a11c3eb3a16490d609d